### DISCUSSION TITLE
fastapi-users full example breaks when switching to PostgreSQL

### DISCUSSION THREAD

**Discussion by betterprompts** [fastapi-users/fastapi-users#1284] [Q&A]
I'm a noob exploring fastapi-users and trying to get a working version running with PostgreSQL. This is where the documentation's full code example usually helps tremendously. The full example is located here:
https://fastapi-users.github.io/fastapi-users/12.1/configuration/full-example/

However, this code example seems to break the moment I change the DB from SQLite to PostgreSQL. Specifically when I change:
DATABASE_URL = "postgresql+asyncpg://user:password@host:port/name"

It looks like the issue has to do with changes in sqlalchemy 2.0. 

I managed to update db.py to fix these changes, but it just results in other errors.

Is it possible to provide a full working example (for regular and OAuth2 versions) in the documentation not just for SQLite, but for PostgreSQL also since these don't appear to be easily swappable?

Thanks in advance from someone new and exploring fastapi-users!

**Comment by frankie567** [comment]
Hi @betterprompts 👋 

I've several projects running FastAPI Users with PostgreSQL without any particular issue. Two things to check:

* Make sure your DATABASE_URL is correct (in particular username and password)
* Make sure you installed `asyncpg` driver dependency (ref: https://fastapi-users.github.io/fastapi-users/12.1/configuration/databases/sqlalchemy/#asynchronous-driver)

If that doesn't solve your problem, share with us the error you're getting (with the full stack trace) so we can understand better what's going on.

**Comment by betterprompts** [comment]
After changing only one line of code in the OAuth2 full example:
`DATABASE_URL = "sqlite+aiosqlite:///name.db"`
to 
`DATABASE_URL = "postgresql+asyncpg://user:password@host:port/name"`
(of course using actual valid PostgreSQL database credentials)

I get this error:
`sqlalchemy.exc.ArgumentError: Type annotation for "User.oauth_accounts" can't be correctly interpreted for Annotated Declarative Table form.  ORM annotations should normally make use of the ``Mapped[]`` generic type, or other ORM-compatible generic type, as a container for the actual type, which indicates the intent that the attribute is mapped. Class variables that are not intended to be mapped by the ORM should use ClassVar[].  To allow Annotated Declarative to disregard legacy annotations which don't use Mapped[] to pass, set "__allow_unmapped__ = True" on the class or a superclass this class. (Background on this error at: https://sqlalche.me/e/20/zlpr)`

I fixed the sqlalchemy Mapped[] issue in db.py (although admittedly, not fully understanding all the code) and I can successfully register a user (write) to the PostgreSQL DB (asyncpg was installed), but then I get other errors. 

For example, for the regular example, I get this error:
`AttributeError: 'User' object has no attribute 'is_active'`

And for the OAuth2 example, I get this error:
`sqlalchemy.exc.InvalidRequestError: Class <class 'app.db.OAuthAccount'> does not have a __table__ or __tablename__ specified and does not inherit from an existing table-mapped class.`

I realize some of these errors may be trivial, but I thought it would be better to ask to see if there's just a full working example that I can look at instead of spending a lot of time fixing errors that result in more errors that need fixing.

**Reply by frankie567** [reply]
The full example provided should run as if no matter the DB ORM. I've just checked and it works.

Can you run the command `pip freeze` and copy/paste the output? It looks like some errors may be provoked by outdated dependencies.

Also, make sure you look at the latest example (for version 12.1): https://fastapi-users.github.io/fastapi-users/12.1/configuration/full-example/

**Comment by betterprompts** [comment]
Yes, you are right. I must have been using an older version of the documentation without realizing it... and it was frustrating me out of my mind!

I re-applied the full example code like you said and it works. 

Sorry for the inconvenience and thanks for your help!
